### PR TITLE
[PfemFluid] New frictional viscoplastic thermal element

### PR DESCRIPTION
**Pull request by AFranci**
A frictional viscoplastic thermal constitutive model is added. Moreover, the constitutive models are reorganised and thermal models are collected in separate folders.

### CODE REVIEW

**Review by rlrangel — DISMISSED** [review]

File: applications/PfemFluidDynamicsApplication/custom_constitutive/fluid_laws/temperature_dependent/frictional_viscoplastic_temperature_dependent_2D_law.cpp [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+
+        return 0;
+    }
+
+    double FrictionalViscoplasticTemperatureDependent2DLaw::GetEffectiveViscosity(ConstitutiveLaw::Parameters &rParameters) const
+    {
+        return rParameters.GetConstitutiveMatrix()(2, 2);
+    }
+
+    double FrictionalViscoplasticTemperatureDependent2DLaw::GetEffectiveDensity(ConstitutiveLaw::Parameters &rParameters) const
```
Comment (AlejandroCornejo) [inline applications/PfemFluidDynamicsApplication/custom_constitutive/fluid_laws/temperature_dependent/frictional_viscoplastic_temperature_dependent_2D_law.cpp:97]: honestly, you can unify these methods by adding an rVariable as an input in the method. Then this method should be named `GetEffectiveVariable(rParams, rVariable)`

File: applications/PfemFluidDynamicsApplication/custom_constitutive/fluid_laws/temperature_dependent/frictional_viscoplastic_temperature_dependent_2D_law.cpp [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+            effective_viscosity = this->GetValueFromTable(TEMPERATURE, DYNAMIC_VISCOSITY, rParameters);
+        }
+        else
+        {
+            effective_viscosity = r_properties[DYNAMIC_VISCOSITY];
+        }
+        return effective_viscosity;
+    }
+
+    double FrictionalViscoplasticTemperatureDependent2DLaw::GetEffectiveFrictionAngle(ConstitutiveLaw::Parameters &rParameters) const
```
Comment (AlejandroCornejo) [inline applications/PfemFluidDynamicsApplication/custom_constitutive/fluid_laws/temperature_dependent/frictional_viscoplastic_temperature_dependent_2D_law.cpp:127]: same... Only one method to do the same

File: applications/PfemFluidDynamicsApplication/custom_constitutive/fluid_laws/temperature_dependent/frictional_viscoplastic_temperature_dependent_2D_law.h [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+    ///@}
+    ///@name Protected Operators
+    ///@{
+
+    ///@}
+    ///@name Protected Operations
+    ///@{
+
+    /// Get the effective viscosity (in dynamic units -- Pa s) for the fluid.
+    double GetEffectiveViscosity(ConstitutiveLaw::Parameters& rParameters) const override;
```
Comment (AlejandroCornejo) [inline applications/PfemFluidDynamicsApplication/custom_constitutive/fluid_laws/temperature_dependent/frictional_viscoplastic_temperature_dependent_2D_law.h:119]: only one..

File: applications/PfemFluidDynamicsApplication/custom_constitutive/fluid_laws/temperature_dependent/frictional_viscoplastic_temperature_dependent_3D_law.cpp [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+        KRATOS_ERROR << "Incorrect or missing COHESION provided in process info for FrictionalViscoplasticTemperatureDependent3DLaw: "
+                     << rMaterialProperties[COHESION] << std::endl;
+    }
+
+    if (rMaterialProperties[ADAPTIVE_EXPONENT] < 0.0) {
+        KRATOS_ERROR << "Incorrect or missing ADAPTIVE_EXPONENT provided in process info for FrictionalViscoplasticTemperatureDependent3DLaw: "
+                     << rMaterialProperties[ADAPTIVE_EXPONENT] << std::endl;
+    }
+
+    if (rMaterialProperties[BULK_MODULUS] <= 0.0) {
```
Comment (AlejandroCornejo) [inline applications/PfemFluidDynamicsApplication/custom_constitutive/fluid_laws/temperature_dependent/frictional_viscoplastic_temperature_dependent_3D_law.cpp:77]: you can use `KRATOS_ERROR_IF`, looks cleaner

File: applications/PfemFluidDynamicsApplication/custom_constitutive/fluid_laws/temperature_dependent/frictional_viscoplastic_temperature_dependent_3D_law.cpp [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+    }
+
+    return 0;
+}
+
+double FrictionalViscoplasticTemperatureDependent3DLaw::GetEffectiveViscosity(ConstitutiveLaw::Parameters& rParameters) const {
+    return rParameters.GetConstitutiveMatrix()(5, 5);
+}
+
+double FrictionalViscoplasticTemperatureDependent3DLaw::GetEffectiveDensity(ConstitutiveLaw::Parameters& rParameters) const {
```
Comment (AlejandroCornejo) [inline applications/PfemFluidDynamicsApplication/custom_constitutive/fluid_laws/temperature_dependent/frictional_viscoplastic_temperature_dependent_3D_law.cpp:89]: same triplication of methods

Comment (AFranci) [reply applications/PfemFluidDynamicsApplication/custom_constitutive/fluid_laws/temperature_dependent/frictional_viscoplastic_temperature_dependent_3D_law.cpp:89]: OK, thanks! I'll take a look next week, in the meanwhile I close the PR.

**Pull request closed without merging**